src/checkers/inference/model/ConstraintManager.java

```diff
@@ -167,6 +165,17 @@ private AnnotationLocation getCurrentLocation() {
         }
     }
 
+    // visitorState.getPath().getLeaf() caused NullPointerException on some real project
+    // and doesn't always return correct source tree. Using class tree at least shows which
+    // Java source class there is such unsatisfiable error and also doesn't throw exception
+    private Tree getCurrentTreeLocation() {
+        if (visitorState.getPath() != null) {
+            return visitorState.getPath().getLeaf();
+        } else {
+            return visitorState.getClassTree();
```

Comment:
Add TODOs for both methods - why hitting this case(else branch)?

Reply:
Issue: find a minimum test case
Throw NPE in else branch, and find the project that causes exception.